product-service/src/functions/get-product-by-id/handler.ts

```diff
@@ -0,0 +1,29 @@
+import type { ValidatedEventAPIGatewayProxyEvent } from "@libs/api-gateway";
+import { formatJSONResponse } from "@libs/api-gateway";
+import { middyfy } from "@libs/lambda";
+import { books } from "../../content/books.json";
+
+const getProductById: ValidatedEventAPIGatewayProxyEvent<any> = async (
+  event
+) => {
+  const productId = event.pathParameters?.productId;
+
+  if (!productId) {
+    return formatJSONResponse({
+      status: 400,
```

Comment:
If you trying to implement REST, you should move status code out of body. In OpenAI you describe this situation as code 400 but you got 200 instead